onnxruntime/core/mlas/lib/q4_dq.cpp

```diff
@@ -1047,16 +985,307 @@ MlasDequantizeBlockwise(
     }
 }
 
-template
+template void
+MlasDequantizeBlockwise<float, 4>(float* dst,
+                                  const uint8_t* src,
+                                  const float* scales,
+                                  const uint8_t* zero_points,
+                                  int block_size,
+                                  bool columnwise,
+                                  int rows,
+                                  int columns,
+                                  MLAS_THREADPOOL* thread_pool);
+
+#ifdef MLAS_JBLAS
+
+template <typename T>
+static size_t
+JblasQ4BuSize(int block_size, size_t N, size_t K, bool isAsym)
+{
```

Comment:
Could you please put the Jblas based implementation in a separate file for maintenance?

Reply:
already moved to jblas_gemm.cpp